lightclient/src/background.rs

```diff
@@ -138,26 +154,28 @@ impl<TPlatform: PlatformRef, TChain> BackgroundTask<TPlatform, TChain> {
                 sender,
                 chain_id,
             } => {
-                let id = self.next_id(chain_id);
+                let (data, client) = self.for_chain_id(chain_id);
```

Comment:
I would probably rename data to `chain_state` or something.

data is too generic